src/PgRvn.Server/RqlQuery.cs

```diff
@@ -91,128 +92,85 @@ private ICollection<PgColumn> GenerateSchema()
+                    pgType = output switch
                     {
-                        case DateTime dt:
-                            size = 8;
-                            type = PgTypeOIDs.Timestamp;
-                            if (dt.Kind == DateTimeKind.Utc)
-                                type = PgTypeOIDs.TimestampTz;
-                            break;
-
-                        case DateTimeOffset:
-                            type = PgTypeOIDs.TimestampTz;
-                            size = 8;
-                            break;
-
-                        case TimeSpan:
-                            type = PgTypeOIDs.Interval;
-                            size = 16;
-                            break;
-                        default:
-                            break;
-                    }
+                        DateTime dt => (dt.Kind == DateTimeKind.Utc) ? PgTimestampTz.Default : PgTimestamp.Default,
+                        DateTimeOffset => PgTimestampTz.Default,
+                        TimeSpan => PgInterval.Default,
+                        _ => pgType
+                    };
                 }
 
-                Columns[prop.Name] = new PgColumn
-                {
-                    Name = prop.Name,
-                    FormatCode = resultsFormat,
-                    TypeModifier = -1,
-                    TypeObjectId = type,
-                    DataTypeSize = (short)size,
-                    ColumnIndex = (short)Columns.Count,
-                    TableObjectId = 0
-                };
+                Columns.TryAdd(prop.Name, new PgColumn(prop.Name, (short)Columns.Count, pgType, resultsFormat, -1));
             }
 
-            //Columns["@metadata"] = new PgColumn
-            //{
-            //    Name = "@metadata",
-            //    FormatCode = resultsFormat,
-            //    TypeModifier = -1,
-            //    TypeObjectId = PgTypeOIDs.Text,
-            //    DataTypeSize = -1,
-            //    ColumnIndex = (short)Columns.Count,
-            //    TableObjectId = 0
-            //};
-
-
-            Columns["json()"] = new PgColumn
+            if (Columns.TryGetValue("json()", out var jsonColumn))
             {
-                Name = "json()",
-                FormatCode = resultsFormat,
-                TypeModifier = -1,
-                TypeObjectId = PgTypeOIDs.Json,
-                DataTypeSize = -1,
-                ColumnIndex = (short)Columns.Count,
-                TableObjectId = 0
-            };
+                jsonColumn.PgType = PgJson.Default;
+            }
+            else
+            {
+                Columns["json()"] = new PgColumn("json()", (short)Columns.Count, PgJson.Default, resultsFormat);
+            }
 
             if (_result.Includes.Count != 0)
             {
                 _hasIncludes = true;
-                Columns["is_include()"] = new PgColumn
+
+                if (Columns.TryGetValue("is_include()", out var includesColumn))
```

Comment:
What is `is_include()`? Where does it come from?

Reply:
can be removed

Reply:
Done